### PR TITLE
[MOOV-1924]: Added Payment Request Attempt Capture

### PR DESCRIPTION
**Pull request by agranillonf**
Added full and partial capture functionality to payment request transactions/attempts.

This PR requires an updated version of the MoneyMoov Client (>= v0.0.9).

### CODE REVIEW

**Review by Publicker — APPROVED** [review]
Looks good - it'll get a bit tricker to one of us when merging, depending on which PR merges first, if this one or #85.

As loads files have been largely modified, particularly on classes.

**Review by donalnofrixion — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
It doesn't matter which quantity I put, it displays 0.00 as the captured quantity. If I try to make another capture again, it tells me "bad request".
We should remove the button if no more capture is possible on this payment.

https://github.com/nofrixion/nofrixion.webcomponents/assets/127946572/3b8699e8-8a65-4f98-a2e6-58721ac2c9ff

**Review by Publicker — CHANGES_REQUESTED** [review]
All looking good, but found two things.

- The "Total payment was X.XX" label is align to the middle and seems a bit off, but that's just my opinion, maybe @pabloi-nf could have a better take in here
Screenshot 2023-07-14 at 15 31 13

------

- Immediately after capturing, the modal closes and goes back to the PR Details, which is perfect, but it shows the captured value as "0", rather than the captured amount. Note: after reloading the page, it correctly shows the captured value.
Screenshot 2023-07-14 at 15 31 32

**Comment by agranillonf** [comment]
@pabloi-nf @Publicker The captured amount showing as 0 has been fixed.

**Review by Publicker — APPROVED** [review]

**Review by pabloi-nf — APPROVED** [review]

**Comment by agranillonf** [comment]
Won't work until [MoneyMoov.Dotnet #160](https://github.com/nofrixion/moneymoov-dotnet/pull/160) and [NoFrixion.Clients #11](https://github.com/nofrixion/nofrixion.clients/pull/11) are merged, and a new version of the NoFrixion.Clients library (0.0.9) is released to NPM and referenced accordingly.

File: src/utils/parsers.ts [hunk 1]
```diff
@@ -184,21 +188,36 @@ const remotePaymentRequestToLocalPaymentRequest = (remotePaymentRequest: Payment
...
             attemptKey: attemptKey,
             occurredAt: new Date(settledAt ?? authorisedAt ?? 0),
             paymentMethod: walletName
               ? parseWalletNameToPaymentMethodType(walletName)
               : parseApiPaymentMethodTypeToLocalMethodType(paymentMethod),
-            amount: settledAmount > 0 ? settledAmount : authorisedAmount,
+            amount: attemptedAmount,
             currency: currency,
             processor: walletName ? parseApiWalletTypeToLocalWalletType(walletName) : undefined,
+            needsCapture:
```
Comment (sauravmaiti22) [inline src/utils/parsers.ts:209]: To determine this we can check if AuthorizedAmount in attempt is greater than SettledAmount when payment method is card.

**Comment by agranillonf** [comment]
Needs [MoneyMoov.Dotnet #163](https://github.com/nofrixion/moneymoov-dotnet/pull/163) merged to work.

**Review by sauravmaiti22 — APPROVED** [review]

**Review by Publicker — APPROVED** [review]

**Pull request merged**